iceoryx_posh/include/iceoryx_posh/internal/popo/used_chunk_list.hpp

```diff
@@ -21,161 +22,117 @@
-
-                // remove index from used list
-                if (current == m_usedListHead)
-                {
-                    m_usedListHead = m_list[current];
-                }
-                else
-                {
-                    m_list[previous] = m_list[current];
-                }
-
-                // insert index to free list
-                m_list[current] = m_freeListHead;
-                m_freeListHead = current;
-
-                //@todo can we do this cheaper with a global fence in cleanup?
-                m_synchronizer.clear(std::memory_order_release);
-                return true;
-            }
-            previous = current;
+            static_assert(sizeof(RelativePointerData) <= 8U, "The RelativePointerData size must not exceed 64 bit!");
+            static_assert(std::is_trivially_copyable<RelativePointerData>::value,
+                          "The RelativePointerData must be trivially copyable!");
         }
-        return false;
-    }
-
-    // only once from runtime context
-    void setup()
-    {
-        m_synchronizer.test_and_set(std::memory_order_acquire);
-        /// @todo maybe also call init here?
-    }
-
-    // from RouDi context once the applications walked the plank
-    void cleanup()
-    {
-        m_synchronizer.test_and_set(std::memory_order_acquire);
-
-        for (auto& data : m_data)
+        constexpr RelativePointerData(uint16_t id, uint64_t offset) noexcept
+            : m_idAndOffset(static_cast<uint64_t>(id) | (offset << 16U))
         {
-            if (data != nullptr)
-            {
-                mepoo::SharedChunk{data};
-            }
+            cxx::Ensures(offset < MAX_OFFSET && "offset must not exceed MAX:OFFSET!");
         }
 
-        init(); // just to save us from the future self
-    }
-
-  private:
-    void init()
-    {
-        // build list
-        for (uint32_t i = 0u; i < Size; ++i)
+        uint16_t id() const noexcept
         {
-            m_list[i] = i + 1u;
+            return static_cast<uint16_t>(m_idAndOffset & MAX_ID);
         }
 
-        if (Size > 0)
+        uint64_t offset() const noexcept
         {
-            m_list[Size - 1u] = InvalidIndex; // just to save us from the future self
+            return (m_idAndOffset >> 16) & MAX_OFFSET;
         }
-        else
+
+        void reset() noexcept
         {
-            m_list[0u] = InvalidIndex;
+            *this = LOGICAL_NULLPTR;
         }
 
-
-        m_usedListHead = InvalidIndex;
-        m_freeListHead = 0u;
-
-        // clear data
-        for (auto& data : m_data)
+        bool isLogicalNullptr() const noexcept
         {
-            data = nullptr;
+            return m_idAndOffset == LOGICAL_NULLPTR;
         }
 
-        m_synchronizer.clear(std::memory_order_release);
-    }
-
-    bool freeSpaceInList()
-    {
-        return (m_freeListHead != InvalidIndex);
-    }
+        static constexpr uint64_t MAX_ID{std::numeric_limits<uint16_t>::max()};
```

Comment:
Could MAX_ID maybe defined as uint16_t here? Then you have not to do the static_cast in `id()`